test/Person.test.js

```diff
@@ -0,0 +1,192 @@
+    alice = await user(accounts[0], 'alice', 'ALC');
+    bob = await user(accounts[1], 'bob', 'BOB');
+    carol = await user(accounts[2], 'carol', 'CRL');
+
+    // everyone is rich :)
+    await increaseTimeTo((await latestTime()) + duration.years(10));
+  });
+
+  describe('updateExchangeInput', () => {
+    it('reverts if called by non owner', async () => {
+      assertRevert(
+        alice.person.updateExchangeInput(bob.token.address, true, {
+          from: bob.address
+        })
+      );
+    });
+
+    it('updates isEligableExchangeInput mapping', async () => {
+      (await alice.person.isEligableExchangeInput(
+        bob.token.address
+      )).should.equal(false);
+
+      await alice.person.updateExchangeInput(bob.token.address, true, {
+        from: alice.address
+      });
+
+      (await alice.person.isEligableExchangeInput(
+        bob.token.address
+      )).should.equal(true);
+    });
+  });
+
+  describe('updateExchangeOutput', () => {
+    it('updateExchangeOutput reverts if called by non owner', async () => {
+      assertRevert(
+        alice.person.updateExchangeOutput(bob.token.address, true, {
+          from: bob.address
+        })
+      );
+    });
+
+    it('updates isEligableExchangeOutput mapping', async () => {
+      (await alice.person.isEligableExchangeOutput(
+        bob.token.address
+      )).should.equal(false);
+
+      await alice.person.updateExchangeOutput(bob.token.address, true, {
+        from: alice.address
+      });
+
+      (await alice.person.isEligableExchangeOutput(
+        bob.token.address
+      )).should.equal(true);
+    });
+  });
+
+  describe('isExchangeApproved', () => {
+    const isApproved = async () => {
+      return await alice.person.isExchangeApproved(
+        bob.token.address,
+        carol.token.address
+      );
+    };
+
+    it('returns true when exchange is approved', async () => {
+      (await isApproved()).should.equal(false);
+
+      await alice.person.updateExchangeInput(bob.token.address, true, {
+        from: alice.address
+      });
+
+      (await isApproved()).should.equal(false);
+
+      await alice.person.updateExchangeOutput(carol.token.address, true, {
+        from: alice.address
+      });
+
+      (await isApproved()).should.equal(true);
+    });
+  });
+
+  describe('exchangeTransfer', () => {
+    it('can spend one hop through the trust graph', async () => {
+      // A -> B -> C
+      // TODO: Actual code doing this would package all of these transactions
+      //  Into a script contract, so that they execute atomically.
+      //  Right now when alice gives bob's exchange permission to withdraw,
+      //  Anyone can race to swap her tokens
+
+      // Bob Trusts Alice
+      await bob.person.updateExchangeInput(alice.token.address, true, {
+        from: bob.address
+      });
+      await bob.person.updateExchangeOutput(bob.token.address, true, {
+        from: bob.address
+      });
+
+      let scriptSrc = `
```

Comment:
So that is what ends up happening here, it is just deployed through the proxy itself. You are right though, we will _probably_ want to create some sort of "default" looking transitive transaction script that takes arrays of addresses to reduce gas costs of deploying a new script every time. I still think it makes sense to have this code here to be compiled inline, so our tests for this person contract aren't dependent on the "default script" contract.

Reply:
It also is how we will cover the use case of transitive transactions that have to call another contract at the end, instead of just depositing into some account.